File: frontend/src/app/Services/User/user.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { RoleUser } from 'src/app/models/role-user.enum';


@Injectable({
  providedIn: 'root'
})
export class UserService {
  
  private baseUrl = 'http://localhost:8080/api/User';

  constructor(private http:HttpClient) {  }
  public Login(user:any){
    return this.http.post<any>(`${this.baseUrl}/login`, user);
  }

  public signUp(user: any,roleUser:RoleUser) {
    return this.http.post(`${this.baseUrl}/signUp/${roleUser}`, user);
  }

  activateAccount(validationDto: any): Observable<boolean> {
    return this.http.put<boolean>(`${this.baseUrl}/activateAccount`, validationDto);
  }

  forgetPassword(user: any): Observable<boolean> {
    return this.http.post<boolean>(`${this.baseUrl}/forgetPassword`, user);
  }

  updatePassword(validationDto: any, newPassword: string): Observable<boolean> {
    return this.http.put<boolean>(`${this.baseUrl}/forgetPassword/${newPassword}`, validationDto);
  }
}

File: backend/src/main/java/com/youcode/MonSupplier/services/interfaces/UserService.java
package com.youcode.MonSupplier.services.interfaces;

import com.youcode.MonSupplier.models.Dtos.UserDto.UserDto;
import com.youcode.MonSupplier.models.Dtos.ValidationDto.ValidationDto;
import com.youcode.MonSupplier.models.Enums.RoleUser;

import java.util.Optional;

public interface UserService {
    Optional Login (UserDto userDto);
    Boolean activateAccount (ValidationDto validationDto);

    Boolean sendCodeForgetPassword(UserDto userDto);

    Boolean updatePassword(ValidationDto validationDto, String newPassword);

    Boolean signUp(Object userObject, RoleUser roleUser);
}
